### ISSUE TITLE
Aves "Loads" photos everytime the app is opened and scans countries as well.

### ISSUE THREAD

**Issue by petazonk** [deckerst/aves#414]
I loaded about 100k of photos onto my phone as I love using Aves to look at the statistics, tags and map. I've found that now whenever I open Aves it can take up to 15 seconds to finish loading, mapping and cataloging photos even when I haven't taken any photos.

Whenever I open Aves it now has a big delay before I can start looking at all the photos. If I wait for it to finish, then switch apps (swipe up from the bottom and then press the Aves Camera Icon on the home screen), as soon as I switch back again it goes through Loading, Countries and Locating.

It would be nice if when I opened Aves in the Camera folder it was ready nearly straight away. I understand that having so many photos on my phone means that it will slow down a bit but shouldn't the camera folder be more responsive?

**Comment by deckerst** [comment]
"Loading" everything everytime is normal (or at least, by design).

"Cataloguing" should happen only once (unless you have new items).

"Locating" should also happen once, but it's possible that you have items with coordinates for which the Geocoder cannot find an address. In that case, the app cannot know whether there is no address or the Geocoder simply failed, so Aves retries everytime. For example I have such pictures, taken in the Ha Long bay (coordinates but no address). Maybe you have such pictures too.

I'm aware that loading takes more time when you have more items. That's the price to pay to get then instant filtering/stats/navigation in the app. To be honest, I'm focused on providing a decent loading time for ~20k items, which is amply enough for most people. Instantly opening 100k items while keeping Aves instant features seems impossible to me (at least for now with the current approach).

But I'm very much interested to dig further about the repeated "Cataloguing", which shouldn't happen.

**Comment by petazonk** [comment]
You are correct about Cataloguing (I double checked) as it only appears with new photos. Loading does take a while due to such a large number of photos (definitely not the norm).

For the locating issue, no address, would it be possible to add them to a "No Address" tag so that it doesn't attempt to keep doing it?

I have the bulk of my photos in a custom directory, eg /photos. So, is there some trick such that it doesn't attempt to scan this directory everytime or is it more a case that the data structures are large due to the 100k of photos so it will always have a lag now?

Is it possible to have an archive type feature such that when you mark a directory as archived, it assumes that no further changes will take place unless a rescan is initiated? Maybe a new virtual TAG could be created called Archive?

**Comment by SIRSteiner** [comment]
Here it takes more than 10 seconds to load at start with 27k images. Other galleries don't need such time.

**Comment by petazonk** [comment]
I've dropped my photos to about 25k and it takes about five seconds to
finish the start process.

**Comment by deckerst** [comment]
Not in the near future.

**Comment by SIRSteiner** [comment]
7 month later... 😉 Did you think about it again?

**Comment by deckerst** [comment]
@SIRSteiner Please do not bump issues if you have nothing to add.

**Comment by SIRSteiner** [comment]
Is it not legitimate to ask about progress on this performance problem? 
If yes, please give an update. If not feel free to close this issue. But this would be a pitty.

**Comment by deckerst** [comment]
> Is it not legitimate to ask about progress on this performance problem?

No, it's not. You can subscribe to issues to get notifications on new messages. No need to ask or whine.

**Comment by SRaresP** [comment]
> > Is it not legitimate to ask about progress on this performance problem?
> 
> No, it's not. You can subscribe to issues to get notifications on new messages. No need to ask or whine.

completely backwards way of managing user sentiment.

**Comment by gagarwa** [comment]
> > > Is it not legitimate to ask about progress on this performance problem?
> > 
> > 
> > No, it's not. You can subscribe to issues to get notifications on new messages. No need to ask or whine.
> 
> completely backwards way of managing user sentiment.

I disagree, I think it's perfectly fair, especially for a complex performance issue.  If others have something to offer, they can contribute as well.
